tests/index.test.js

```diff
@@ -0,0 +1,108 @@
+import nock from 'nock';
+
+import fetchRedditPostsByTopic, { fetchJson, extractFields, getRedditUrl } from '../lib/index.js';
+
+describe('fetchRedditPostsByTopicPromise', () => {
+
+  describe('Fetch json', () => {
+
+    const response = { 
+      data: { 
+        children: [
+          { data: { id: 1, title: 'RTX 3080 Crashes', thumbnail: 'self', subreddit: 'Nvidia', ups: 635, downs: 34 } },
+          { data: { id: 2, title: 'Cyberpunk 2077 goes gold', thumbnail: 'self', subreddit: 'Nvidia', ups: 1124, downs: 450 } },
+          { data: { id: 3, title: 'PS5 release data', thumbnail: 'self', subreddit: 'Nvidia', ups: 12903 , downs: 2415 } }
+        ] 
+      } 
+    };
+
+    it('Should match returned data', async () => {
+      const redditUrl = 'https://reddit.com/r/pics/search.json?q=Gaming&sort=new';
+
+      const responseData = { data: { children: [{ title: 'RTX 3080 Crashes' }] } };
+
+      nock('https://reddit.com')
+        .get('/r/pics/search.json?q=Gaming&sort=new')
+        .reply(200, responseData);
+
+      return await fetchJson(redditUrl)
+        .then(res => expect(res).toEqual(responseData));
+    });
+
+    it('Should return a certain response when only topic is passed', async () => {
+      const { data: { children } } = response;
+
+      nock('https://www.reddit.com')
+        .get('/r/pics/search.json?q=Gaming&sort=new')
+        .reply(200, response);
+
+      const request = await fetchRedditPostsByTopic('Gaming');
+
+      expect(request).toEqual(children);
+    });
+
+    it('Should return a certain response when topic and fields is passed', async () => {
+      const fields = ['id', 'title'];
+
+      const { data: { children } } = response;
+
+      nock('https://www.reddit.com')
+        .get('/r/pics/search.json?q=Gaming&sort=new')
+        .reply(200, response);
+
+      const request = await fetchRedditPostsByTopic('Gaming', fields);
+
+      const expectedResponse = extractFields(children, fields);
```

Comment:
Are we testing extract fields here or fetchRedditPostsByTopic? If fetchRedditPostsByTopic we should not rely on extractFields. Instead creating another mock object to compare against.

Something like `mockFetchRedditPostByTopicResponse`

Reply:
I used extractFields just so I don't have to create the mock object since it would be the same thing.